pkg/cloud/externalconn/connection.go

```diff
@@ -21,6 +21,15 @@ import (
 // an External Connection object. This interface should expose read-only
 // methods that are required to interact with the External Connection object.
 type ExternalConnection interface {
+	// CreateConnectionStatement returns a `CREATE EXTERNAL CONNECTION` statement
+	// that is functionally equivalent to the statement that created the external
+	// connection in the first place.
+	//
+	// NB: The returned string will contain unredacted secrets and should not be
+	// persisted.
+	CreateConnectionStatement() string
```

Comment:
nit: maybe build "unredacted" into the name? `UnredactedConnectionStatement() string`? Also feels a bit weird to call this a "Create" when it returns a string

Reply:
good point, changed.